forge-gui-mobile/src/forge/adventure/scene/DuelScene.java

```diff
@@ -336,14 +360,23 @@ public void enter() {
 
         hostedMatch = MatchController.hostMatch();
 
-        GameRules rules = new GameRules(GameType.Constructed);
+        GameRules rules;
+
+        if (eventData != null){
+            rules = new GameRules(eventData.eventRules.gameType);
+            rules.setGamesPerMatch(eventData.eventRules.gamesPerMatch);
+            bossBattle = false;
+        }
+        else{
+            rules = new GameRules(GameType.Adventure);
+            rules.setGamesPerMatch(1);
+        }
         rules.setPlayForAnte(false);
         rules.setMatchAnteRarity(true);
-        rules.setGamesPerMatch(1);
         rules.setManaBurn(false);
         rules.setWarnAboutAICards(false);
 
-        hostedMatch.setEndGameHook(() -> DuelScene.this.GameEnd());
+        //hostedMatch.setEndGameHook(() -> DuelScene.this.GameEnd());
```

Comment:
What will happen if adventurewinlose is skipped via user settings? does the gameend is called somewhere else?